roles/dcos_agent_windows/tasks/main.yml

```diff
@@ -8,23 +13,27 @@
     path: '{{ base_windows_dir }}\etc\dcos-version.json'
   register: previous_installation
 
+- name: Windows | Create dcos_installation.fact.ps1
+  win_copy:
+    content: get-content {{ base_windows_dir }}\etc\ansible\facts.d\dcos_installation.fact -Raw | ConvertFrom-Json
```

Comment:
the content of dcos_installaton.fact file will be available after task "Copy dcos-version.json to fact directory" . Therefore I don't recommend line 16 .. 22 to be added with anything